### PR TITLE
 f-registration@0.26.0 Validation changes

### PR DESCRIPTION
**Pull request by adamashton**
### Changed
- Validation for inputs now displayed when focus is lost

### Added
- Email input max length is now 50 characters
- Password input must be between 4 and 50 characters

## UI Review Checks
- [X] README and/or UI Documentation has been [created|updated]
- [X] Unit tests have been [created|updated]
- [X] This code has been checked with regard to [our accessibility standards](http://fozzie.just-eat.com/documentation/general/accessibility/checklist)

## Browsers Tested

- [X] Chrome (latest)
- [X] Mobile - Pixel 2 - Chrome

### CODE REVIEW

File: packages/f-registration/src/components/tests/Registration.test.js [hunk 1]
```diff
@@ -315,6 +334,26 @@ describe('Registration', () => {
...
+                lastNameInput.trigger('blur');
+                await flushPromises();
+
+                // Assert
+                expect(wrapper.vm.shouldShowLastNameMaxLengthError).toBe(true);
+            } finally {
+                wrapper.destroy();
+            }
+        });
+
```
Comment (ashleynolan) [inline packages/f-registration/src/components/tests/Registration.test.js:453]: Is it worth having tests for the email and password fields too for when focus is lost?

Comment (adamashton) [reply packages/f-registration/src/components/tests/Registration.test.js:453]: Sure 👍

**Review by minrwhite — APPROVED** [review]

File: packages/f-registration/src/components/Registration.vue [hunk 2]
```diff
@@ -112,6 +114,13 @@
                         <warning-icon :class="$style['o-form-error-icon']" />
                         Please enter a valid email address
                     </p>
+                    <p
+                        v-if="shouldShowEmailMaxLengthError"
+                        :class="$style['o-form-error']"
+                        data-test-id='error-email-maxlength'>
+                        <warning-icon :class="$style['o-form-error-icon']" />
+                        Email exceeds 50 characters
```
Comment (philgoff) [inline packages/f-registration/src/components/Registration.vue]: This should read **Email address exceeds 50 characters** according to the ticket.

Comment (philgoff) [reply packages/f-registration/src/components/Registration.vue]: Question: I know this ticket says UK only but has this been set up for translations in the future?

Comment (adamashton) [reply packages/f-registration/src/components/Registration.vue]: Thanks Phil, good spot.

Yes, it's begun here. https://github.com/justeat/fozzie-components/blob/master/packages/f-registration/src/tenants/en-GB.js

**Review by philgoff — APPROVED** [review]
Approving now that we have another ticket in sprint coming to enable translations.

**Pull request merged**